### PR TITLE
Add Forecast CSC

### PR DESCRIPTION
**Pull request by couger01**

### CODE REVIEW

File: sal_interfaces/Forecast/Forecast_Telemetry.xml [hunk 1]
```diff
@@ -0,0 +1,674 @@
...
+      <Count>360</Count>
+      <Description>Forecast time</Description>
+      <Units>unitless</Units>
+      <EFDB_Name>forecast_time</EFDB_Name>
+    </item>
+    <item>
+      <IDL_Type>float</IDL_Type>
+      <Count>360</Count>
+      <Description>Forecast temperature</Description>
+      <Units>deg</Units>
```
Comment (wvreeven) [inline sal_interfaces/Forecast/Forecast_Telemetry.xml:71]: The unit for temperature is `deg_C`. Please change here and everywhere else.

File: sal_interfaces/Forecast/Forecast_Telemetry.xml [hunk 2]
```diff
@@ -0,0 +1,674 @@
...
+      <IDL_Type>double</IDL_Type>
+      <Count>360</Count>
+      <Description>Forecast time</Description>
+      <Units>unitless</Units>
+      <EFDB_Name>forecast_time</EFDB_Name>
+    </item>
+    <item>
+      <IDL_Type>float</IDL_Type>
+      <Count>360</Count>
+      <Description>Forecast temperature</Description>
```
Comment (wvreeven) [inline sal_interfaces/Forecast/Forecast_Telemetry.xml:70]: It is clear that this is the `Forecast` temperature so you can drop that prefix here and everywhere else.

File: sal_interfaces/Forecast/Forecast_Telemetry.xml [hunk 3]
```diff
@@ -0,0 +1,674 @@
...
+      <Count>360</Count>
+      <Description>Forecast wind direction</Description>
+      <Units>deg</Units>
+      <EFDB_Name>wind_direction</EFDB_Name>
+    </item>
+    <item>
+      <IDL_Type>long</IDL_Type>
+      <Count>360</Count>
+      <Description>Forecast sea level pressure</Description>
+      <Units>meter*pascal</Units>
```
Comment (wvreeven) [inline sal_interfaces/Forecast/Forecast_Telemetry.xml:120]: That's a strange unit for pressure. Are you sure it isn't `Pascal` instead?

File: sal_interfaces/Forecast/Forecast_Telemetry.xml [hunk 4]
```diff
@@ -0,0 +1,674 @@
...
+      <Count>360</Count>
+      <Description>Forecast relative humidity</Description>
+      <Units>pct</Units>
+      <EFDB_Name>relative_humidity</EFDB_Name>
+    </item>
+    <item>
+      <IDL_Type>float</IDL_Type>
+      <Count>360</Count>
+      <Description>Forecast ghi backwards</Description>
+      <Units>watt*hour*meter**2</Units>
```
Comment (wvreeven) [inline sal_interfaces/Forecast/Forecast_Telemetry.xml:134]: Shouldn't this be `watt/hour/m**2`? Please review this and all other units that have `*` and that should have `/` instead.

File: sal_interfaces/Forecast/Forecast_Telemetry.xml [hunk 5]
```diff
@@ -0,0 +1,674 @@
...
+      <IDL_Type>float</IDL_Type>
+      <Count>360</Count>
+      <Description>Forecast evapotransipiration</Description>
+      <Units>unitless</Units>
+      <EFDB_Name>evapo_transipiration</EFDB_Name>
+    </item>
+    <item>
+      <IDL_Type>float</IDL_Type>
+      <Count>360</Count>
+      <Description>Forecast reference evaporated transipiration fao</Description>
```
Comment (wvreeven) [inline sal_interfaces/Forecast/Forecast_Telemetry.xml:259]: What is `fao`?

File: sal_interfaces/Forecast/Forecast_Telemetry.xml [hunk 6]
```diff
@@ -0,0 +1,674 @@
...
+      <Units>unitless</Units>
+      <EFDB_Name>reference_evapo_transipiration_fao</EFDB_Name>
+    </item>
+  </SALTelemetry>
+  <SALTelemetry>
+    <Subsystem>Forecast</Subsystem>
+    <EFDB_Topic>Forecast_trend_day</EFDB_Topic>
```
Comment (wvreeven) [inline sal_interfaces/Forecast/Forecast_Telemetry.xml:266]: Please rename this to `Forecast_dailyTrend`. Note that all comments, that apply to the items of `Forecast_hourlyTrend`, apply to the items of `Forecast_dailyTrend`.

**Review by wvreeven — CHANGES_REQUESTED** [review]
Please review my comments and make changes accordingly.

File: sal_interfaces/Forecast/Forecast_Telemetry.xml [hunk 7]
```diff
@@ -0,0 +1,674 @@
...
+      <IDL_Type>float</IDL_Type>
+      <Count>360</Count>
+      <Description>Forecast snow fraction</Description>
+      <Units>pct</Units>
+      <EFDB_Name>snow_fraction</EFDB_Name>
+    </item>
+    <item>
+      <IDL_Type>long</IDL_Type>
+      <Count>360</Count>
+      <Description>Forecast pictocode</Description>
```
[the remark this answers is not in the document]
Comment (couger01) [reply sal_interfaces/Forecast/Forecast_Telemetry.xml:168]: You know the forecast symbols that are used? That's apparently a pictocode.

File: sal_interfaces/Forecast/Forecast_Telemetry.xml [hunk 8]
```diff
@@ -0,0 +1,674 @@
...
+  </SALTelemetry>
+  <SALTelemetry>
+    <Subsystem>Forecast</Subsystem>
+    <EFDB_Topic>Forecast_trend_1h</EFDB_Topic>
+    <item>
+      <IDL_Type>double</IDL_Type>
+      <Count>360</Count>
+      <Description>Forecast time</Description>
+      <Units>unitless</Units>
+      <EFDB_Name>forecast_time</EFDB_Name>
```
[the remark this answers is not in the document]
Comment (wvreeven) [reply sal_interfaces/Forecast/Forecast_Telemetry.xml:65]: I forgot about that. Then `timestamp` please.

File: sal_interfaces/Forecast/Forecast_Telemetry.xml (hunk 7 above)
[the remark this answers is not in the document]
Comment (wvreeven) [reply sal_interfaces/Forecast/Forecast_Telemetry.xml:168]: Can you add that to the description?

File: sal_interfaces/Forecast/Forecast_Telemetry.xml (hunk 4 above)
Comment (couger01) [reply sal_interfaces/Forecast/Forecast_Telemetry.xml:134]: Unit is Whm-2.

Comment (wvreeven) [reply sal_interfaces/Forecast/Forecast_Telemetry.xml:134]: So then it should be `watt*hour/m**2` so please correct that and check all other units as well.

File: sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml [hunk 9]
```diff
@@ -26,649 +26,649 @@
...
+      <EFDB_Name>modelrun</EFDB_Name>
+      <Description>The time that the forecast was run in UTC.</Description>
       <IDL_Type>string</IDL_Type>
       <Units>unitless</Units>
       <Count>1</Count>
     </item>
     <item>
-      <EFDB_Name>modelrun_updatetime_utc</EFDB_Name>
+      <EFDB_Name>modelrunUpdatetime</EFDB_Name>
       <Description>The time that the model was last updated</Description>
```
Comment (wvreeven) [inline sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml:51]: Please add here that the time is in UTC.

File: sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml [hunk 10]
```diff
@@ -26,649 +26,649 @@
...
   <SALTelemetry>
-    <Subsystem>Forecast</Subsystem>
-    <EFDB_Topic>Forecast_trend_1h</EFDB_Topic>
+    <Subsystem>WeatherForecast</Subsystem>
+    <EFDB_Topic>WeatherForecast_hourlyTrend</EFDB_Topic>
     <item>
       <IDL_Type>double</IDL_Type>
       <Count>360</Count>
-      <Description>Forecast time</Description>
+      <Description>time</Description>
```
Comment (wvreeven) [inline sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml:63]: Time in UTC? If yes, please add that.

File: sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml [hunk 11]
```diff
@@ -26,649 +26,649 @@
...
       <Units>pct</Units>
-      <EFDB_Name>precipitation_probability</EFDB_Name>
+      <EFDB_Name>precipitationProbability</EFDB_Name>
     </item>
     <item>
       <IDL_Type>long</IDL_Type>
       <Count>360</Count>
-      <Description>Forecast cape</Description>
+      <Description>cape</Description>
       <Units>Joule*kilogram</Units>
```
Comment (wvreeven) [inline sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml:239]: This unit should be Joule/kilogram

File: sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml [hunk 12]
```diff
@@ -26,649 +26,649 @@
...
   <SALTelemetry>
-    <Subsystem>Forecast</Subsystem>
-    <EFDB_Topic>Forecast_trend_day</EFDB_Topic>
+    <Subsystem>WeatherForecast</Subsystem>
+    <EFDB_Topic>WeatherForecast_dailyTrend</EFDB_Topic>
     <item>
       <IDL_Type>double</IDL_Type>
       <Count>14</Count>
-      <Description>Forecast time</Description>
+      <Description>time</Description>
```
Comment (wvreeven) [inline sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml:270]: Time in UTC? If yes please indicate here.

File: sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml [hunk 13]
```diff
@@ -26,649 +26,649 @@
...
       <Units>unitless</Units>
-      <EFDB_Name>trend_time</EFDB_Name>
+      <EFDB_Name>timestamp</EFDB_Name>
     </item>
     <item>
       <IDL_Type>long</IDL_Type>
       <Count>14</Count>
-      <Description>Forecast pictocode</Description>
+      <Description>pictocode</Description>
```
Comment (wvreeven) [inline sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml:277]: Please update the description to indicate what this means.

File: sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml [hunk 14]
```diff
@@ -26,649 +26,649 @@
...
       <Units>deg</Units>
-      <EFDB_Name>wind_direction</EFDB_Name>
+      <EFDB_Name>windDirection</EFDB_Name>
     </item>
     <item>
       <IDL_Type>long</IDL_Type>
       <Count>14</Count>
-      <Description>Forecast maximum sea level pressure</Description>
+      <Description>maximum sea level pressure</Description>
       <Units>bar</Units>
```
Comment (wvreeven) [inline sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml:369]: `bar` is not an SI unit. Please change to Pascal here and everywhere else. Sorry, I missed this during the first pass.

File: sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml [hunk 15]
```diff
@@ -26,649 +26,649 @@
...
+      <EFDB_Name>predictabilityClass</EFDB_Name>
     </item>
     <item>
       <IDL_Type>long</IDL_Type>
       <Count>14</Count>
-      <Description>Forecast maximum total cloud cover</Description>
+      <Description>maximum total cloud cover</Description>
       <Units>pct</Units>
-      <EFDB_Name>totalcloudcover_max</EFDB_Name>
+      <EFDB_Name>totalcloudcoverMax</EFDB_Name>
```
Comment (wvreeven) [inline sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml:426]: Please change to `totalCloudCoverMax`.

File: sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml [hunk 16]
```diff
@@ -26,649 +26,649 @@
...
     </item>
     <item>
       <IDL_Type>long</IDL_Type>
       <Count>14</Count>
-      <Description>Forecast minimum total cloud cover</Description>
+      <Description>minimum total cloud cover</Description>
       <Units>pct</Units>
-      <EFDB_Name>totalcloudcover_min</EFDB_Name>
+      <EFDB_Name>totalcloudcoverMin</EFDB_Name>
```
Comment (wvreeven) [inline sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml:433]: Please change to `totalCloudCoverMin`.

File: sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml [hunk 17]
```diff
@@ -26,649 +26,649 @@
...
     </item>
     <item>
       <IDL_Type>long</IDL_Type>
       <Count>14</Count>
-      <Description>Forecast mean total cloud cover</Description>
+      <Description>mean total cloud cover</Description>
       <Units>pct</Units>
-      <EFDB_Name>totalcloudcover_mean</EFDB_Name>
+      <EFDB_Name>totalcloudcoverMean</EFDB_Name>
```
Comment (wvreeven) [inline sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml:440]: Please change to `totalCloudCoverMean`.

File: sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml [hunk 18]
```diff
@@ -26,649 +26,649 @@
...
     </item>
     <item>
       <IDL_Type>long</IDL_Type>
       <Count>14</Count>
-      <Description>Forecast spread total cloud cover</Description>
+      <Description>spread total cloud cover</Description>
       <Units>pct</Units>
-      <EFDB_Name>totalcloudcover_spread</EFDB_Name>
+      <EFDB_Name>totalcloudcoverSpread</EFDB_Name>
```
Comment (wvreeven) [inline sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml:447]: Please change to `totalCloudCoverSpread`.

File: sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml [hunk 19]
```diff
@@ -26,649 +26,649 @@
...
     </item>
     <item>
       <IDL_Type>long</IDL_Type>
       <Count>14</Count>
-      <Description>Forecast total extra terrestrial radiation</Description>
-      <Units>watt*hour/minute**2</Units>
-      <EFDB_Name>extraterrestrialradiation_total</EFDB_Name>
+      <Description>total extra terrestrial radiation</Description>
+      <Units>watt*hour/meter**2</Units>
+      <EFDB_Name>extraterrestrialradiationTotal</EFDB_Name>
```
Comment (wvreeven) [inline sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml:461]: Please change to `extraTerrestrialRadiationTotal`.

File: sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml [hunk 20]
```diff
@@ -26,649 +26,649 @@
...
+      <EFDB_Name>gustMean</EFDB_Name>
     </item>
     <item>
       <IDL_Type>long</IDL_Type>
       <Count>14</Count>
-      <Description>Forecast maximum low clouds</Description>
+      <Description>maximum low clouds</Description>
       <Units>pct</Units>
-      <EFDB_Name>lowclouds_max</EFDB_Name>
+      <EFDB_Name>lowcloudsMax</EFDB_Name>
```
Comment (wvreeven) [inline sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml:489]: Please change to `lowCloudsMax`.

File: sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml [hunk 21]
```diff
@@ -26,649 +26,649 @@
...
     </item>
     <item>
       <IDL_Type>long</IDL_Type>
       <Count>14</Count>
-      <Description>Forecast minimum low clouds</Description>
+      <Description>minimum low clouds</Description>
       <Units>pct</Units>
-      <EFDB_Name>lowclouds_min</EFDB_Name>
+      <EFDB_Name>lowcloudsMin</EFDB_Name>
```
Comment (wvreeven) [inline sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml:496]: Please change to `lowCloudsMin`.

File: sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml [hunk 22]
```diff
@@ -26,649 +26,649 @@
...
     </item>
     <item>
       <IDL_Type>long</IDL_Type>
       <Count>14</Count>
-      <Description>Forecast mean low clouds</Description>
+      <Description>mean low clouds</Description>
       <Units>pct</Units>
-      <EFDB_Name>lowclouds_mean</EFDB_Name>
+      <EFDB_Name>lowcloudsMean</EFDB_Name>
```
Comment (wvreeven) [inline sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml:503]: Please change to `lowCloudsMean`.

Comment (wvreeven) [reply sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml:503]: And apply similar lower camel casing to midcloudsXXX and hicloudsXXX as well.

File: sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml [hunk 23]
```diff
@@ -26,649 +26,649 @@
...
+      <Description>mean high clouds</Description>
       <Units>pct</Units>
-      <EFDB_Name>hiclouds_mean</EFDB_Name>
+      <EFDB_Name>hicloudsMean</EFDB_Name>
     </item>
     <item>
       <IDL_Type>long</IDL_Type>
       <Count>14</Count>
-      <Description>Forecast sunshine time</Description>
+      <Description>sunshine time</Description>
```
Comment (wvreeven) [inline sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml:550]: Time in UTC? Please indicate and make sure all other times indicate as well.

File: sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml [hunk 24]
```diff
@@ -26,649 +26,649 @@
...
     </item>
     <item>
       <IDL_Type>float</IDL_Type>
       <Count>14</Count>
-      <Description>Forecast maximum skin temperature</Description>
-      <Units>deg</Units>
-      <EFDB_Name>skintemperature_max</EFDB_Name>
+      <Description>maximum skin temperature</Description>
+      <Units>deg_C</Units>
+      <EFDB_Name>skintemperatureMax</EFDB_Name>
```
Comment (wvreeven) [inline sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml:580]: Please change all `skintemperature` to `skinTemperature`.

File: sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml [hunk 25]
```diff
@@ -26,649 +26,649 @@
...
     </item>
     <item>
       <IDL_Type>float</IDL_Type>
       <Count>14</Count>
-      <Description>Forecast minimum dew point temperature</Description>
-      <Units>deg</Units>
-      <EFDB_Name>dewpointtemperature_min</EFDB_Name>
+      <Description>dew point temperature</Description>
+      <Units>deg_C</Units>
+      <EFDB_Name>dewpointtemperatureMin</EFDB_Name>
```
Comment (wvreeven) [inline sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml:608]: Please change all `dewpointtemperatue` to `dewPointTemperature`.

File: sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml [hunk 26]
```diff
@@ -26,649 +26,649 @@
...
+      <EFDB_Name>capeMean</EFDB_Name>
     </item>
     <item>
       <IDL_Type>float</IDL_Type>
       <Count>14</Count>
-      <Description>Forecast maximum lifted index</Description>
+      <Description>maximum lifted index</Description>
       <Units>unitless</Units>
-      <EFDB_Name>liftedindex_max</EFDB_Name>
+      <EFDB_Name>liftedindexMax</EFDB_Name>
```
Comment (wvreeven) [inline sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml:643]: Please change all `liftedindex` to `liftedIndex`.

**Review by wvreeven — CHANGES_REQUESTED** [review]
Much better, thanks. However, there still is improvement possible. Also you're mixing descriptions that start with a capital letter and end in a dot with descriptions that start with a lower case letter and end without a dot. Please make sure that all start with a capital letter and end in a dot. Indicate for all times that they are in UTC. Make sure to use proper lower camel casing everywhere (I indicated several in my comments).

File: sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml [hunk 27]
```diff
@@ -0,0 +1,674 @@
...
+      <IDL_Type>float</IDL_Type>
+      <Count>360</Count>
+      <Description>Evapotransipiration.</Description>
+      <Units>unitless</Units>
+      <EFDB_Name>evapoTransipiration</EFDB_Name>
+    </item>
+    <item>
+      <IDL_Type>float</IDL_Type>
+      <Count>360</Count>
+      <Description>Reference evaporated transipiration fao.</Description>
```
Comment (wvreeven) [inline sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml:259]: Please explain the meaning of `fao`.

File: sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml [hunk 28]
```diff
@@ -0,0 +1,674 @@
...
+      <IDL_Type>long</IDL_Type>
+      <Count>14</Count>
+      <Description>Mean high clouds.</Description>
+      <Units>pct</Units>
+      <EFDB_Name>hiCloudsMean</EFDB_Name>
+    </item>
+    <item>
+      <IDL_Type>long</IDL_Type>
+      <Count>14</Count>
+      <Description>Sunshine time (utc).</Description>
```
Comment (wvreeven) [inline sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml:550]: `utc` -> `UTC`.

File: sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml [hunk 29]
```diff
@@ -0,0 +1,674 @@
...
+      <IDL_Type>float</IDL_Type>
+      <Count>14</Count>
+      <Description>Mean lifted index.</Description>
+      <Units>unitless</Units>
+      <EFDB_Name>liftedIndexMean</EFDB_Name>
+    </item>
+    <item>
+      <IDL_Type>float</IDL_Type>
+      <Count>14</Count>
+      <Description>Evapotranspiration.</Description>
```
Comment (wvreeven) [inline sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml:662]: Please change to `Evaporation transpiration.`.

File: sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml [hunk 30]
```diff
@@ -0,0 +1,674 @@
...
+      <Units>millimeter</Units>
+      <EFDB_Name>evapoTranspiration</EFDB_Name>
+    </item>
+    <item>
+      <IDL_Type>float</IDL_Type>
+      <Count>14</Count>
+      <Description>Reference evapotranspiration as defined by FAO(Food and agriculture UN organization) using Penman-Monteith equation.</Description>
```
Comment (wvreeven) [inline sal_interfaces/WeatherForecast/WeatherForecast_Telemetry.xml:669]: Copy this description to the other `referenceEvapoTranspirationFao` and add a space before `(` in both descriptions.

**Review by wvreeven — APPROVED** [review]
Looks good enough. A bit meager on the descriptions ;) Please take my remaining comments into account and rebase the branch against `develop` before merging.

**Pull request merged**